connectors/src/connectors/microsoft/lib/graph_api.ts

```diff
@@ -1,21 +1,58 @@
 import type { Result } from "@dust-tt/types";
 import { assertNever, Err, Ok } from "@dust-tt/types";
 import type { Client } from "@microsoft/microsoft-graph-client";
+import { GraphError } from "@microsoft/microsoft-graph-client";
 import type * as MicrosoftGraph from "@microsoft/microsoft-graph-types";
 
 import type { MicrosoftNode } from "@connectors/connectors/microsoft/lib/types";
 import {
   internalIdFromTypeAndPath,
   typeAndPathFromInternalId,
 } from "@connectors/connectors/microsoft/lib/utils";
+import { ExternalOAuthTokenError } from "@connectors/lib/error";
+import logger from "@connectors/logger/logger";
+
+export async function clientApiGet(client: Client, endpoint: string) {
+  try {
+    return await client.api(endpoint).get();
+  } catch (error) {
+    logger.error({ error, endpoint }, `Graph API call threw an error`);
+    if (
+      error instanceof GraphError &&
+      error.message.includes("Access denied")
+    ) {
+      throw new ExternalOAuthTokenError(error);
+    }
+    throw error;
+  }
+}
+
+export async function clientApiPost(
+  client: Client,
+  endpoint: string,
+  content: any // eslint-disable-line @typescript-eslint/no-explicit-any
+) {
+  try {
+    return await client.api(endpoint).post(content);
+  } catch (error) {
+    logger.error({ error, endpoint }, `Graph API call threw an error`);
+    if (
+      error instanceof GraphError &&
+      error.message.includes("Access denied")
+    ) {
+      throw new ExternalOAuthTokenError(error);
```

Comment:
Are we sure there's no transient access denied from those kind of API calls, and/or that they are about the whole MS access, not just one resource?
Asking because IIRC we're moving forward to pause connectors that get ExtOAuthTokenError, which would be an issue if the answer to the question above is not yes

Reply:
Unfortunately I can't answer "yes" for sure here :-) I've added logs so that we can see the exact error when it happens, so could be fine tuned if needed. But I doubt we can receive an transient "Access denied" error.

Reply:
Ok good for me if at least in the observed case it was not transient